src/esp/sim/Simulator.h

```diff
@@ -716,6 +719,21 @@ class Simulator {
                              scene::SceneNode& agentParentNode);
   agent::Agent::ptr addAgent(const agent::AgentConfiguration& agentConfig);
 
+  /**
+   * @brief Initialize sensor and attach to sceneNode of a particular object
+   * with uuid objectId
+   * @param objectId    Id of the object to which a sensor will be initialized
+   * at its node
+   * @param position  vec3f of position to set spec to
+   * @param orientation vec3f or orientation to set spec to
+   * @param resolution vec2i of resolution to set spec to
+   *
+   */
+  void addSensorToObject(const int objectId,
```

Comment:
When designing this API, please take a look at the definitions of the other similar "add" functions, such as `addAgent`.

The return can be a pointer to that new sensor. So you can memorize this sensor in your demo. You do not need that global sensorSuite_ in the simulator.
The input should be object Id or a `scene::SceneNode` of that object (You can create an overload function, and use one to call the other.) + a sensorSpec (instead of the pos + orientation + resolution).